src/Microsoft.CodeQuality.Analyzers/UnitTests/QualityGuidelines/ValidateArgumentsOfPublicMethodsTests.cs

```diff
@@ -3768,7 +3768,9 @@ private void M2(C c)
         var x = c._field;
     }
 }
-", CompilerDiagnostics.None,
+",
+            // Test0.cs(8,9): error CS0165: Use of unassigned local variable 'c2'
+            DiagnosticResult.CompilerError("CS0165").WithLocation(8, 9),
```

Comment:
💭 I'd almost prefer to see `WithMessage` used here _instead of the comment.

```suggestion
            DiagnosticResult.CompilerError("CS0165").WithLocation(8, 9).WithMessage("Use of unassigned local variable 'c2'"),
```

Reply:
See https://github.com/dotnet/roslyn-analyzers/pull/3005#discussion_r343703984 for the main discussion on this change.